src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -193,7 +198,29 @@ private ResultSet executeQueryHelper(String querySql, boolean unlimitedBillingBy
                   .get(projectId, qr.getJobReference().getJobId())
                   .setLocation(qr.getJobReference().getLocation())
                   .execute();
+
+          BiEngineStatistics biEngineStatistics = null;
+          if (referencedJob != null) {
+            JobStatistics2 statistics2 = referencedJob.getStatistics().getQuery();
+            if (statistics2 != null) {
+              biEngineStatistics = statistics2.getBiEngineStatistics();
+            }
+          }
+
+          if (biEngineStatistics != null) {
+            biEngineMode = biEngineStatistics.getBiEngineMode();
```

Comment:
Similar comment to the other block, this could get moved into the one path that you actually set `biEngineStatistics` above.